core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -50,6 +57,37 @@ object ZkUtils extends Logging {
   val EntityConfigPath = "/config"
   val EntityConfigChangesPath = "/config/changes"
 
+  /** true if java.security.auth.login.config is set to some jaas file which has "Client" entry. **/
+
+  val isSecure: Boolean = {
+    val loginConfigurationFile: String = System.getProperty("java.security.auth.login.config")
+    var isSecurityEnabled  = false
+    if (loginConfigurationFile != null && loginConfigurationFile.length > 0) {
+      val config_file: File = new File(loginConfigurationFile)
+      if (!config_file.canRead) {
+        throw new KafkaException("File " + loginConfigurationFile + " cannot be read.")
```

Comment:
Kafka usually uses `format` instead of String concatenation. Now that we require Scala 2.10, I personally favour String interpolation (ie s"File $logingConfigurationFile cannot be read."), but a question has been raised on a separate PR about this. I'll update this comment once Jun replies on that other PR.